daemon/logger/jsonfilelog/read.go

```diff
@@ -171,9 +178,21 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 	}
 	defer func() {
 		f.Close()
+		fileWatcher.Remove(name)
 		fileWatcher.Close()
 	}()
 
+	logWatcherCtx, logWatcherClosed := context.WithCancel(ctx)
```

Comment:
We need one to notify that the watcher is closed, and the other to notify that the goroutine needs to exit.